File: csharp/src/WorkflowRunner.Starknet/Models/StarknetPublishTransactionRequest.cs
﻿using Train.Solver.Blockchain.Abstractions.Models;

namespace Train.Solver.Blockchain.Starknet.Models;

public class StarknetPublishTransactionRequest : BaseRequest
{
    public string FromAddress { get; set; } = null!;

    public string CallData { get; set; } = null!;

    public string Nonce { get; set; } = null!;

    public Fee Fee { get; set; } = null!;
}

File: js/src/Blockchain/Blockchain.Starknet/Models/StarknetPublishTransactionRequest .ts
import { BaseRequest } from "../../Blockchain.Abstraction/Models/BaseRequest";
import { Fee } from "../../Blockchain.Abstraction/Models/FeesModels/Fee";

export interface StarknetPublishTransactionRequest extends BaseRequest {
    fromAddress: string;
    callData: string;
    nonce: string;
    fee: Fee;
  }